server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -315,6 +322,31 @@ public class AccountManagerImpl extends ManagerBase implements AccountManager, M
     private int _cleanupInterval;
     private List<String> apiNameList;
 
+    protected static Map<String, UserTwoFactorAuthenticator> userTwoFactorAuthenticationProvidersMap = new HashMap<>();
+
+    private List<UserTwoFactorAuthenticator> userTwoFactorAuthenticationProviders;
+
+    static ConfigKey<Boolean> enableUserTwoFactorAuthentication = new ConfigKey<Boolean>("Advanced",
+            Boolean.class,
+            "enable.user.two.factor.authentication",
+            "false",
+            "Determines whether two factor authentication is enabled or not. This can also be configured at domain level.",
+            false,
```

Comment:
@harikrishna-patnala 
these settings are dynamic , right ?

Reply:
These can be made dynamic, I'll make it now. I wanted the reviewer's opinion, so kept it false then. I'll change it.